autosklearn/data/target_validator.py

```diff
@@ -35,37 +31,42 @@ class TargetValidator(BaseEstimator):
         is_classification: bool
             A bool that indicates if the validator should operate in classification mode.
             During classification, the targets are encoded.
-        encoder: typing.Optional[BaseEstimator]
+        encoder: Optional[BaseEstimator]
             Host a encoder object if the data requires transformation (for example,
             if provided a categorical column in a pandas DataFrame)
-        enc_columns: typing.List[str]
+        enc_columns: List[str]
             List of columns that where encoded
     """
     def __init__(self,
                  is_classification: bool = False,
-                 logger: typing.Optional[PickableLoggerAdapter] = None,
+                 logger: Optional[PickableLoggerAdapter] = None,
                  ) -> None:
         self.is_classification = is_classification
 
-        self.data_type = None  # type: typing.Optional[type]
+        self.data_type = None  # type: Optional[type]
 
-        self.encoder = None  # type: typing.Optional[BaseEstimator]
+        # NOTE: type update
+        #   Encoders don't seems to have a unified base class for
+        #   the methods 'transform'. Could make a `prototype` class that
+        #   duck types for the 'transform', 'fit_transform' methods
+        #   This is however fine while we only use ordinal encoder
+        self.encoder = None  # type: Optional[OrdinalEncoder]
 
-        self.out_dimensionality = None  # type: typing.Optional[int]
-        self.type_of_target = None  # type: typing.Optional[str]
+        self.out_dimensionality = None  # type: Optional[int]
+        self.type_of_target = None  # type: Optional[str]
 
         self.logger = logger if logger is not None else logging.getLogger(__name__)
 
         # Store the dtype for remapping to correct type
-        self.dtype = None  # type: typing.Optional[type]
+        self.dtype = None  # type: Optional[Type]
 
         self._is_fitted = False
 
     def fit(
         self,
-        y_train: SUPPORTED_TARGET_TYPES,
-        y_test: typing.Optional[SUPPORTED_TARGET_TYPES] = None,
-    ) -> BaseEstimator:
+        y_train: Union[list, np.ndarray, pd.Series, pd.DataFrame],
```

Comment:
There's functionally no difference but with some googling, I think it's best to convert it to the `typing.List` type.

I'll fix this.

Reply:
Turns out this is prevalent throughout the code, I will only fix this in the code files I touched.

Reply:
Hm, now I'm wondering - you also changed the code occurrences of `list` to `List` while I only thought about the occurrences in the type hints. Is there a reason for that?

Reply:
I couldn't find any info online but here's two points I could come up with.
* `List` is a more general form of list, while it's unlikely to matter, it doesn't hurt. In general we should probably be checking for a `Sequence` type which supports list like iteration and access.
* It leaves `list` as the operation that converts things to a list when searching, rather than acting as both a type and an operation, e.g. when searching for where we convert something to a list, we wont have to also look through places where we use it for `isinstance` type checking.

A quick try in IPython:
```python
l = [1,2,3]
from typing import List
isinstance(l, List) # True

# However this doesn't work
isinstance(l, List[int]) # Error
```

From the `typing.py` module, it appears it's just an `_alias` with extended type support with a type `T`
```python
List = _alias(list, T, inst=False)
```